distance-metrics/src/simple.rs

```diff
@@ -0,0 +1,201 @@
+use super::metric::Metric;
```

Comment:
diff distance-metrics/src/simple.rs ../qdrant/lib/segment/src/spaces/simple.rs 

```diff
8,9c8,9
< use super::types::VectorElementType;
< use super::types::ScoreType;
---
> use crate::data_types::vectors::VectorElementType;
> use crate::types::{Distance, ScoreType};
31c31,35
<     fn distance(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
---
>     fn distance() -> Distance {
>         Distance::Euclid
>     }
> 
>     fn similarity(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
69c73,77
<     fn distance(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
---
>     fn distance() -> Distance {
>         Distance::Dot
>     }
> 
>     fn similarity(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
107c115,119
<     fn distance(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
---
>     fn distance() -> Distance {
>         Distance::Cosine
>     }
> 
>     fn similarity(v1: &[VectorElementType], v2: &[VectorElementType]) -> ScoreType {
165c177
<         1.0 - score
---
>         score
```

Reply:
In the future, I usually prefer reading unified diffs (`diff -u`).